From 249b04060a1175b970bdca9abf5e09e64247deb1 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Thu, 27 Mar 2014 13:54:08 +0000
Subject: [PATCH 173/351] regulator: arizona-micsupp: Only process DT if no
 pdata

In arizona-core we only process the DT if no pdata is present, we should
be consistent in this approach. This patch adds similar handling into
this driver.

Change-Id: Icfd8bc4446007c22588655a1490b468fa31f35aa
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/regulator/arizona-micsupp.c |   11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/regulator/arizona-micsupp.c b/drivers/regulator/arizona-micsupp.c
index 7fe081d..5ce6483 100644
--- a/drivers/regulator/arizona-micsupp.c
+++ b/drivers/regulator/arizona-micsupp.c
@@ -281,9 +281,11 @@ static int arizona_micsupp_probe(struct platform_device *pdev)
 	config.driver_data = micsupp;
 	config.regmap = arizona->regmap;
 
-	ret = arizona_micsupp_of_get_pdata(arizona, &config);
-	if (ret < 0)
-		return ret;
+	if (!dev_get_platdata(arizona->dev)) {
+		ret = arizona_micsupp_of_get_pdata(arizona, &config);
+		if (ret < 0)
+			return ret;
+	}
 
 	if (arizona->pdata.micvdd)
 		config.init_data = arizona->pdata.micvdd;
@@ -304,7 +306,8 @@ static int arizona_micsupp_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	arizona_micsupp_of_put_pdata(&config);
+	if (!dev_get_platdata(arizona->dev))
+		arizona_micsupp_of_put_pdata(&config);
 
 	platform_set_drvdata(pdev, micsupp);
 
-- 
1.7.9.5

